### PR TITLE
Add support for `google_compute_instance_group` to TGC

### PR DESCRIPTION
**Pull request by delimaneto**
adding support for compute.googleapis.com/InstanceGroup
```release-note:none
```

### CODE REVIEW

File: mmv1/third_party/tgc/tests/data/example_google_compute_instance_group.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "ancestry_path": "{{.Ancestry}}/project/{{.Provider.project}}",
+        "resource": {
+            "version": "v1",
+            "discovery_document_uri": "https://www.googleapis.com/discovery/v1/apis/compute/v1/rest",
+            "discovery_name": "InstanceGroup",
+            "parent": "//cloudresourcemanager.googleapis.com/projects/{{.Provider.project}}",
+            "data": {
+                "description": "Terraform test instance group",
+                "name": "terraform-test",
+                "zone": "us-central1-a"
```
Comment (sushant00) [inline mmv1/third_party/tgc/tests/data/example_google_compute_instance_group.json:14]: Can we update the conversion code such that the zone returned is of the same format as it is in the CAI list assets:     https://www.googleapis.com/compute/v1/projects/tf-deployer-2/zones/us-central1-a

I think the change should be likely in expandComputeInstanceGroupZone() function.

Comment (delimaneto) [reply mmv1/third_party/tgc/tests/data/example_google_compute_instance_group.json:14]: gcloud asset list --project="tf-deployer-2" --asset-types="[compute.googleapis.com/InstanceGroup](https://www.google.com/url?q=http://compute.googleapis.com/InstanceGroup&sa=D&source=buganizer&usg=AOvVaw3r1w5coaUB4mr5xTz5xvSb)" --content-type="resource" --filter="name: //[compute.googleapis.com/projects/tf-deployer-2/zones/us-central1-a/instanceGroups/terraform-test](https://www.google.com/url?q=http://compute.googleapis.com/projects/tf-deployer-2/zones/us-central1-a/instanceGroups/terraform-test&sa=D&source=buganizer&usg=AOvVaw1tyWZiVL_COHIew-3xKqYO)"
---
ancestors:
- projects/111685897256
- folders/868528268369
- folders/409525994184
- organizations/777838403257
assetType: [compute.googleapis.com/InstanceGroup](https://www.google.com/url?q=http://compute.googleapis.com/InstanceGroup&sa=D&source=buganizer&usg=AOvVaw3r1w5coaUB4mr5xTz5xvSb)
name: //[compute.googleapis.com/projects/tf-deployer-2/zones/us-central1-a/instanceGroups/terraform-test](https://www.google.com/url?q=http://compute.googleapis.com/projects/tf-deployer-2/zones/us-central1-a/instanceGroups/terraform-test&sa=D&source=buganizer&usg=AOvVaw1tyWZiVL_COHIew-3xKqYO)
resource:
  data:
    creationTimestamp: '2024-05-14T08:13:05.619-07:00'
    description: Terraform test instance group
    fingerprint: 42WmSpB8rSM=
    id: '9155869679643705710'
    name: terraform-test
    selfLink: [https://www.googleapis.com/compute/v1/projects/tf-deployer-2/zones/us-central1-a/instanceGroups/terraform-test](https://www.google.com/url?q=https://www.googleapis.com/compute/v1/projects/tf-deployer-2/zones/us-central1-a/instanceGroups/terraform-test&sa=D&source=buganizer&usg=AOvVaw1bMl-SIrsVcBpNgyF8YFRS)
    size: 0
    zone: [https://www.googleapis.com/compute/v1/projects/tf-deployer-2/zones/us-central1-a](https://www.google.com/url?q=https://www.googleapis.com/compute/v1/projects/tf-deployer-2/zones/us-central1-a&sa=D&source=buganizer&usg=AOvVaw32XFT9pRTz6TbaN12fRPC5)
  discoveryDocumentUri: [https://www.googleapis.com/discovery/v1/apis/compute/v1/rest](https://www.google.com/url?q=https://www.googleapis.com/discovery/v1/apis/compute/v1/rest&sa=D&source=buganizer&usg=AOvVaw2wNRkR4WguIg4eIJWzU5Ay)
  discoveryName: InstanceGroup
  location: us-central1-a
  parent: //[cloudresourcemanager.googleapis.com/projects/111685897256](https://www.google.com/url?q=http://cloudresourcemanager.googleapis.com/projects/111685897256&sa=D&source=buganizer&usg=AOvVaw3X9chD-PGsMYfR97yWVSAI)
  version: v1
updateTime: '2024-05-14T15:13:05.660666Z'

The zone on the list is as `us-central1-a`, right?

Comment (delimaneto) [reply mmv1/third_party/tgc/tests/data/example_google_compute_instance_group.json:14]: @sushant00 

Whether you go to the https://www.googleapis.com/discovery/v1/apis/compute/v1/rest and search for "InstanceGroup" object you'll see that the zone returned from there is an url as documented. Are you sure would you like that I change the format?

 ```
 "zone": {
    "description": "[Output Only] The URL of the zone where the instance group is located (for zonal resources).",
    "type": "string"
  },
```

Comment (delimaneto) [reply mmv1/third_party/tgc/tests/data/example_google_compute_instance_group.json]: done, thanks

File: mmv1/third_party/tgc/compute_instance_group.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	"reflect"
+
+	"github.com/GoogleCloudPlatform/terraform-google-conversion/v5/tfplan2cai/converters/google/resources/cai"
+	"github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google-beta/google-beta/transport"
+)
+
+const ComputeInstanceGroupAssetType string = "compute.googleapis.com/InstanceGroup"
+
+func resourceComputeInstanceGroup() cai.ResourceConverter {
```
Comment (roaks3) [inline mmv1/third_party/tgc/compute_instance_group.go:13]: Nit: I think convention is to include "converter", ie. `resourceConverterComputeInstanceGroup`

**Review by roaks3 — APPROVED** [review]
LGTM

File: mmv1/third_party/tgc/compute_instance_group.go (hunk 2 above)
Comment (delimaneto) [reply mmv1/third_party/tgc/compute_instance_group.go]: done, thanks

**Review by sushant00 — APPROVED** [review]

**Review by roaks3 — APPROVED** [review]

**Pull request merged**